### ISSUE TITLE
error: namespace "thrust::cuda" has no member "par_nosync"

### ISSUE THREAD

**Issue by fengshiyu58** [ceres-solver/ceres-solver#974]
/home/fsy/ws/CPPLibs/ceres-solver/internal/ceres/cuda_kernels_bsm_to_crs.cu.cc(160): error: namespace "thrust::cuda" has no member "par_nosync"

/home/fsy/ws/CPPLibs/ceres-solver/internal/ceres/cuda_kernels_bsm_to_crs.cu.cc(159): error: no instance of overloaded function "thrust::inclusive_scan" matches the argument list
            argument types are: (<error-type>, int *, int *, int *)

2 errors detected in the compilation of "/home/fsy/ws/CPPLibs/ceres-solver/internal/ceres/cuda_kernels_bsm_to_crs.cu.cc".
make[2]: *** [internal/ceres/CMakeFiles/ceres_cuda_kernels.dir/build.make:76：internal/ceres/CMakeFiles/ceres_cuda_kernels.dir/cuda_kernels_bsm_to_crs.cu.cc.o]

**Comment by sandwichmaker** [comment]
cc: @DmitriyKorchemkin

**Comment by shantong123** [comment]
i met the same problem, and i have already uploaded to the latest one

**Comment by DmitriyKorchemkin** [comment]
@shantong123 , do you mean "the latest one available from nvidia" (which is 12.0) or "the latest one available from [some] linux distribution"?

Please provide us with the version of CUDA toolkit that is being used.

One can find a path to `nvcc` that is being found by CMake in `CMakeCache.txt` (in build directory) by searching for a string  `CMAKE_CUDA_COMPILER`, and running this `nvcc` executable with argument `--version` will provide version of the toolkit:

```
> nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Wed_Sep_21_10:33:58_PDT_2022
Cuda compilation tools, release 11.8, V11.8.89
Build cuda_11.8.r11.8/compiler.31833905_0
```

**Comment by sandwichmaker** [comment]
Closing because lack of feedback.
